app/controllers/api/alerts_statuses_controller.rb

```diff
@@ -0,0 +1,60 @@
+module Api
+  class AlertsStatusesController < BaseController
+    include Subcollections::AlertStatusStates
+
+    def update
+      if @req.action == "providers_alerts"
+        providers_alerts_resource
+      else
+        super
+      end
+    end
+
+    def providers_alerts_resource
+      payload = {"providers" => []}
+      ExtManagementSystem.all.each do |provider|
+        payload["providers"] << {
+          "environment" => "production",
+          "name"        => provider.name,
+          "type"        => provider.class.to_s,
+          "id"          => provider.id}.merge("alerts" => alerts_and_states(provider))
+      end
+      render_resource :alertes_statuses, payload
```

Comment:
as another thought (instead of OPTIONS), how much data is this ?

should this just be a subcollection on providers ?  so you can fetch the alert statuses on a single provider or all providers ?

i.e.

```
GET /api/providers/:id/alert_statuses
```

or

```
GET /api/providers/:id?expand=alert_statuses
```

or for the whole lot:

```
GET /api/providers?expand=resources,alert_statuses
```